### PR TITLE
[Feature](agg-state) support agg state storage array/map/struct

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
## Proposed changes
support agg state storage array/map/struct

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

File: be/src/olap/rowset/segment_v2/column_reader.cpp [hunk 1]
```diff
@@ -120,6 +121,7 @@ Status ColumnReader::create_array(const ColumnReaderOptions& opts, const ColumnM
     if (meta.is_nullable()) {
         array_reader->_sub_readers[2] = std::move(null_reader);
     }
+    array_reader->_meta_type = FieldType::OLAP_FIELD_TYPE_ARRAY;
```
[the remark this answers is not in the document]
Comment (BiteTheDDDDt) [reply be/src/olap/rowset/segment_v2/column_reader.cpp:124]: ColumnReader::new_iterator

**Review by amorynan — APPROVED** [review]
LGTM

**Review by zhangstar333 — APPROVED** [review]

**Pull request merged**